pkgs/tools/networking/connman/connman-notify/default.nix

```diff
@@ -1,38 +1,28 @@
-{ stdenv, fetchFromGitHub, pythonPackages, makeWrapper }:
+{ stdenv, fetchFromGitLab, python3Packages }:
 
-stdenv.mkDerivation {
+python3Packages.buildPythonApplication {
   pname = "connman-notify";
-  version = "2014-06-23";
+  version = "2019-10-05";
 
-  src = fetchFromGitHub {
+  src = fetchFromGitLab {
     owner = "wavexx";
     repo = "connman-notify";
-    rev = "0ed9b5e4a0e1f03c83c4589cabf410cac66cd11d";
-    sha256 = "0lhk417fdg3qxs1marpqp277bdxhwnbyrld9xj224bfk5v7xi4bg";
+    rev = "24b10a51721b54d932f4cd61ef2756423768c015";
+    sha256 = "1k5b5y6114yna2cm0cq82xilicran63hrhlklgv7k3p89057xh8j";
   };
 
-  nativeBuildInputs = [ makeWrapper ];
+  format = "other";
 
-  buildInputs = [
-    pythonPackages.python
-    pythonPackages.dbus-python
-    pythonPackages.pygobject2
-    pythonPackages.pygtk
-    pythonPackages.notify
-  ];
+  pythonPath = with python3Packages; [ dbus-python pygobject3 ];
```

Comment:
I believe we should use propagatedBuildInputs. Though, tbh, nothing is going to depend on application. So it's just a convention to follow because of all the other packages.

Reply:
The section _15.17.2.2.1.1. buildPythonPackage parameters* of the [nixpkg manual](https://nixos.org/nixpkgs/manual/) documents both `pythonPath` and `propagatedBuildInputs`. Many packages in nixpkgs use `pythonPath`. I think it is more appropriate for applications and the convention to use `propagatedBuildInputs` when it is not needed does not look right to me.

Reply:
Ah, I wasn't aware the manual still documented `pythonPath`. Code comments used to say `deprecated`. I think we should document it this way for applications then :+1: 

@FRidh ?